server/aesdsocket.c

```diff
@@ -0,0 +1,529 @@
+
+        // Check for receive errors
+        if (bytes_received == ERROR)
+        {
+            syslog(LOG_ERR, "Receive failed");  
+            s_flags.command_status_success = false;
+            return -1;
+        }
+
+        // Debugging Logs
+        DEBUG_LOG("Buffer : %ld", bytes_received);
+        DEBUG_LOG("Buffer : %s", buf);
+
+        // Write the received data to the file
+        ret_status = write(dataFileDescriptor, buf, bytes_received);
+
+        // Check for write errors
+        if (ret_status == ERROR)
+        {
+            syslog(LOG_ERR, "Writing to file unsuccessful");  
+            s_flags.command_status_success = false;
+            return -1;
+        }
+
+        // Exit condition: Check if the last character in the buffer is a newline
+        if (bytes_received > 0 && buf[bytes_received - 1] == '\n')
+        {
+            break;  // Exit loop if a newline character is encountered
+        }
+    }
+
+    return 0;  
+}
+
+/**
+ * @brief Send the content of a file to the client.
+ * 
+ * This function sends the entire content of the file located at /var/tmp/aesdsocketdata back to the client.
+ *
+ * @return 0 on successful operation, -1 otherwise
+ */
+int send_file_content_to_client(void)
+{
+    ssize_t bytes_transmitted = 0;  // Number of bytes successfully sent to the client
+    char file_buffer[BUF_LEN];  // Buffer to hold file content temporarily
+    ssize_t read_count = 1;  // Number of bytes read from the file (initialize to non-zero to start loop)
+
+    // Reposition the read/write offset of the file to the beginning
+    off_t reposition_status = lseek(dataFileDescriptor, 0, SEEK_SET);
+    if (reposition_status == ERROR)
+    {
+        syslog(LOG_ERR, "lseek operation failed");
+        s_flags.command_status_success = false;
+        return -1;
+    }
+
+    // Loop to handle file reading and sending content
+    while (read_count > 0)  // Loop as long as there's content to read
+    {
+        // Read from the file into the buffer
+        read_count = read(dataFileDescriptor, file_buffer, BUF_LEN);
+        if (read_count == ERROR)
+        {
+            syslog(LOG_ERR, "Reading from file failed");  
+            s_flags.command_status_success = false;
+            return -1;
+        }
+
+        // Debugging information
+        DEBUG_LOG("Bytes read from file : %ld", read_count);
+        DEBUG_LOG("Content in read buffer : \n%s", file_buffer);
+
+        // Transmit the read content to the client
+        bytes_transmitted = send(clientSocketFd, file_buffer, read_count, 0);
+        if (bytes_transmitted == ERROR)
+        {
+            syslog(LOG_ERR, "Transmission to client failed");  
+            s_flags.command_status_success = false;
+            return -1;
+        }
+
+        // Exit condition: Check if the last character in the buffer is a newline
+        if (read_count > 0 && file_buffer[read_count - 1] == '\n')
```

Comment:
Thank you for your feedback. You are right; the check for a newline character isn't necessary in this context, because the loop will naturally exit if read_count <= 0. The additional check for a new line was an oversight and was removed in the updated function version.